packages/nodes-base/nodes/InvoiceNinja/V3/v5/DocumentExecution.ts

```diff
@@ -0,0 +1,111 @@
+import { IBinaryData, IDataObject, IExecuteFunctions, INodeExecutionData, NodeOperationError } from 'n8n-workflow';
+import { invoiceNinjaApiDownloadFile, invoiceNinjaApiRequest, invoiceNinjaApiRequestAllItems, invoiceNinjaApiRequestUploadFile } from '../GenericFunctions';
+
+export const execute = async function (this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+    const items = this.getInputData();
+    const returnData: INodeExecutionData[] = [];
+    const length = items.length;
+    const qs: IDataObject = {};
+
+    let responseData;
+
+    const resource = this.getNodeParameter('resource', 0);
+    const operation = this.getNodeParameter('operation', 0);
+    if (resource !== 'document') throw new Error('Invalid Resource Execution Handler');
+
+    for (let i = 0; i < length; i++) {
+        //Routes: https://github.com/invoiceninja/invoiceninja/blob/v5-stable/routes/api.php or swagger documentation
+        try {
+            if (operation === 'upload') {
+                const uploadRessource = this.getNodeParameter('uploadRessource', i);
+                const uploadRessourceId = this.getNodeParameter('uploadRessourceId', i);
+                const binaryPropertyName = this.getNodeParameter('binaryPropertyName', i);
+                const binaryData: IBinaryData = items[i].binary![binaryPropertyName];
+                if (!binaryData) throw new NodeOperationError(this.getNode(), 'No binary data exists on item!', {
+                    itemIndex: i,
+                });
+                const formData = new FormData();
+                formData.append('documents[]', new Blob([binaryData.data], { type: binaryData.mimeType }), binaryData.fileName);
+                responseData = await invoiceNinjaApiRequestUploadFile.call(
+                    this,
+                    'POST',
+                    `/${uploadRessource}/${uploadRessourceId}/upload?_method=put`,
+                    formData,
+                );
+                responseData = responseData.data;
```

Comment:
@Joffcom can you provide me details / examples, how I can send FormData with a file to the api.
The postman request looks like this:
![image](https://user-images.githubusercontent.com/52678724/217035287-e548fdb8-a184-49db-8926-fcabb9ac6642.png)

I've tried the function this.helpers.requestWithAuthentication and provided eigher property body or property formData. when providing formData to body it failes with the following error: Data after transformation must be a string, an ArrayBuffer, a Buffer, or a Stream. When providing formData as an own property, there was no affect at the ressource in the api, like I'have send no data.

Reply:
Hey @paulwer,

Have you tried checking the HTTP Request node which I think might do this?

Reply:
@Joffcom I found a solution copied by mindee node.
I guess there was something wrong, how the binary was handled while form generation or how the form gets attached, because this solution feeld like an workaround.

Please considder this at your review and maybe improve this section.
If you need any assistence, please reach out.